### ISSUE TITLE
TestReflector_resyncChan is flaky

### ISSUE THREAD

**Issue by brendandburns** [kubernetes/kubernetes#12377]
```
W0807 00:26:38.579625 11537 request.go:308] field selector: v1 - pods - spec.host - : need to check if this is versioned correctly.
W0807 00:26:38.632785 11537 request.go:308] field selector: v1 - pods - spec.host - : need to check if this is versioned correctly.
W0807 00:26:38.890282 11537 request.go:308] field selector: v1 - pods - spec.host - : need to check if this is versioned correctly.
W0807 00:26:38.897578 11537 request.go:308] field selector: v1 - pods - spec.host - : need to check if this is versioned correctly.
E0807 00:26:38.912461 11537 poller.go:63] failed to list: transient error
E0807 00:26:38.945424 11537 poller.go:63] failed to list: transient error
E0807 00:26:38.955091 11537 poller.go:63] failed to list: transient error
E0807 00:26:38.958265 11537 poller.go:63] failed to list: transient error
E0807 00:26:39.046629 11537 poller.go:63] failed to list: transient error
E0807 00:26:39.093522 11537 reflector.go:192] watch of *api.Pod ended with: unexpected object: &{{ } {bar 0 0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} {[] [] <nil> <nil> map[] false []} { [] <nil> []}}
--- FAIL: TestReflector_resyncChan (0.11s)
reflector_test.go:105: resyncChan() is at least 99 milliseconds late??
E0807 00:26:39.325047 11537 reflector.go:245] expected type *api.Pod, but watch event object had type *api.Service
E0807 00:26:39.431736 11537 reflector.go:169] Failed to watch *api.Pod: second watch
E0807 00:26:39.433794 11537 reflector.go:169] Failed to watch *api.Pod: second watch
E0807 00:26:39.434063 11537 reflector.go:128] Failed to list *api.Pod: a list error
E0807 00:26:39.434569 11537 reflector.go:169] Failed to watch *api.Pod: a watch error
E0807 00:26:39.435485 11537 reflector.go:169] Failed to watch *api.Pod: second watch
E0807 00:26:39.446659 11537 reflector.go:192] watch of *api.Pod ended with: very short watch
FAIL
coverage: 75.7% of statements
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache	2.165s
```

**Comment by wojtek-t** [comment]
This is really strange. This test is extremely simple:
- it returns time.NewTimer(time.Millisecond)
- it creates a second channel: time.After(100*time.Millisecond)
- waits for the first element in any channel
It seems to be a problem with the testing infrastructure.

@brendandburns - where does this failure come from? I wasn't able to reproduce it locally.

**Comment by mvdan** [comment]
Practically fixed in 77fd38848584 - undoing the timeout change, I got the error three times out of 10k runs. With the timeout bump to 30s, it seems to never happen.

I guess bumping the timeout isn't the correct way to fix this, though.

@wojtek-t it won't happen if you run one test at a time. Easy to reproduce if you run the test 200 times in parallel, which makes the scheduler be less predictable as the system is stressed.

**Comment by lavalamp** [comment]
(Presumably) fixed by https://github.com/kubernetes/kubernetes/pull/14064
